bee-network/src/peers/manager.rs

```diff
@@ -0,0 +1,564 @@
+                    .await
+                {
+                    if let Err(e) = internal_event_sender
+                        .send(InternalEvent::ReconnectScheduled { peer_id })
+                        .map_err(|_| Error::InternalEventSendFailure("ReconnectScheduled"))
+                    {
+                        warn!("{:?}", e)
+                    }
+                }
+            }
+
+            info!("Reconnector stopped.");
+        });
+
+        trace!("Peer Manager started.");
+
+        Ok(Self::default())
+    }
+}
+
+async fn process_command(
+    command: Command,
+    local_keys: &identity::Keypair,
+    peers: &PeerList,
+    banned_addrs: &BannedAddrList,
+    banned_peers: &BannedPeerList,
+    event_sender: &EventSender,
+    internal_event_sender: &InternalEventSender,
+) -> Result<(), Error> {
+    trace!("Received {:?}.", command);
+
+    match command {
+        Command::AddPeer {
+            id,
+            address,
+            alias,
+            relation,
+        } => {
+            // Note: the control flow seems to violate DRY principle, but we only need to clone `id` in one branch.
+            if relation == PeerRelation::Known {
+                add_peer(id.clone(), address, alias, relation, peers, event_sender).await?;
+
+                let local_keys = local_keys.clone();
+                let peers = peers.clone();
+                let banned_addrs = banned_addrs.clone();
+                let banned_peers = banned_peers.clone();
+                let event_sender = event_sender.clone();
+                let internal_event_sender = internal_event_sender.clone();
+
+                // We automatically connect to known peers. Since we can connect concurrently, we spawn a task here.
+                tokio::spawn(async move {
+                    if let Err(e) = connect_peer(
+                        id,
+                        local_keys,
+                        peers,
+                        banned_addrs,
+                        banned_peers,
+                        internal_event_sender,
+                        event_sender,
+                    )
+                    .await
+                    {
+                        warn!("Failed to connect to peer. Cause: {:?}", e);
+                    }
+                });
+            } else {
+                add_peer(id, address, alias, relation, peers, event_sender).await?;
+            }
+        }
+        Command::RemovePeer { id } => {
+            remove_peer(id, peers, event_sender).await?;
+        }
+        Command::ConnectPeer { id } => {
+            let local_keys = local_keys.clone();
+            let peers = peers.clone();
+            let banned_addrs = banned_addrs.clone();
+            let banned_peers = banned_peers.clone();
+            let event_sender = event_sender.clone();
+            let internal_event_sender = internal_event_sender.clone();
+
+            tokio::spawn(async move {
```

Comment:
I must admit, I'm really not happy with how we're forced to use `tokio::spawn` in a bunch of places to spawn tasks that aren't bound to the runtime at all. It's not your fault of course (I've been forced to do the same) but I think I'm going to work on a better solution to this.